app/models/host/managed.rb

```diff
@@ -734,8 +734,8 @@ def refresh_global_status!
     save!(:validate => false)
   end
 
-  def refresh_statuses
-    HostStatus.status_registry.each do |status_class|
+  def refresh_statuses(which = HostStatus.status_registry)
+    which.each do |status_class|
```

Comment:
Is there any situation where one report import needs to update multiple statuses?